### PR TITLE
Adding tier-2 shell scripts for bz_1994827

### PR DESCRIPTION
**Pull request by rahullepakshi**
Signed-off-by: rlepaksh <rlepaksh@redhat.com>

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Review by sunilkumarn417 — COMMENTED** [review]
Hi @rahullepakshi ,

Added scripts looks good to me and need some changes with file name conventions.
and a pass test result would be good before merge.

File: pipeline/scripts/4/2/tier-2/test-deploy_okr_bz_1994827.sh
Comment (psathyan) [inline pipeline/scripts/4/2/tier-2/test-deploy_okr_bz_1994827.sh:10]: Let's start using the soft link that points to the latest i.e. `rhel-7-latest.yaml`

File: pipeline/scripts/4/2/tier-2/test-deploy_okr_bz_1994827.sh
Comment (psathyan) [inline pipeline/scripts/4/2/tier-2/test-deploy_okr_bz_1994827.sh:17]: Could you use the soft link which points to the latest one? That is, `ibm-vpc-rhel-7-latest.yaml`

File: pipeline/scripts/4/3/tier-2/test-deploy_okr_bz_1994827.sh
Comment (psathyan) [inline pipeline/scripts/4/3/tier-2/test-deploy_okr_bz_1994827.sh:10]: Please make the change here also.

File: pipeline/scripts/4/3/tier-2/test-deploy_okr_bz_1994827.sh
Comment (psathyan) [inline pipeline/scripts/4/3/tier-2/test-deploy_okr_bz_1994827.sh:2]: Kindly fix the doc string... I believe it is `Deploy OKR testing`

File: pipeline/scripts/4/2/tier-2/test-deploy_okr_bz_1994827.sh
Comment (Ameenasuhani) [reply pipeline/scripts/4/2/tier-2/test-deploy_okr_bz_1994827.sh:1]: The file name should include "upgrade" and not "deploy"

it is `***upgrade_okr_bz_1994827.sh`

File: pipeline/scripts/4/3/tier-2/test-deploy_okr_bz_1994827.sh
Comment (Ameenasuhani) [reply pipeline/scripts/4/3/tier-2/test-deploy_okr_bz_1994827.sh:2]: file name should include "upgrade"
it is ***upgrade_okr_bz_1994827.sh

and the doc string could be modified accordingly


and hence modify the doc string accordingly

**Review by Ameenasuhani — CHANGES_REQUESTED** [review]

**Comment by sunilkumarn417** [comment]
Hi @rahullepakshi , please consider some test suites from this PR, https://github.com/red-hat-storage/cephci/pull/868#pullrequestreview-815661832

cc : @Ameenasuhani

**Comment by rahullepakshi** [comment]
I will attach log of test run within today, so this can be merged upon test pass and changes approval

**Comment by rahullepakshi** [comment]
> Hi @rahullepakshi , please consider some test suites from this PR, [#868 (review)](https://github.com/red-hat-storage/cephci/pull/868#pullrequestreview-815661832)
> 
> cc : @Ameenasuhani

@sunilkumarn417 I am thinking to raise a new PR for this

**Comment by veera-raghava-reddy** [comment]
Change the test and script name to the actual usecase in upgrade. Remove OKR in names.

**Comment by rahullepakshi** [comment]
> Change the test and script name to the actual usecase in upgrade. Remove OKR in names.
 @sunilkumarn417 , planning to change tier-2_okr_bz_1994827.yaml to tier-2_upgrade_with-OSDs-up.yaml? If not, please suggest
@Ameenasuhani , planning to change tier-2_upgrade_okr-bz-1876447-rpm.yaml to tier-2_upgrade_with-custom-ceph-cluster.yaml? If ot, please suggest

**Comment by veera-raghava-reddy** [comment]
Also, review if Tier-2 is required in the name? is this used for other tests/scripts?

**Comment by rahullepakshi** [comment]
>> Change the test and script name to the actual usecase in upgrade. Remove OKR in names.



> Also, review if Tier-2 is required in the name? is this used for other tests/scripts?

@veera-raghava-reddy , yes, tier-2 is required in the test suite and configuration files as per naming conventions and is followed from sometime too.

**Comment by sunilkumarn417** [comment]
> @sunilkumarn417 , planning to change tier-2_okr_bz_1994827.yaml to tier-2_upgrade_with-OSDs-up.yaml. If not related, please suggest

@rahullepakshi , the OKR 1994827 scenario is covered in that test suite. But however we can change it to this file name `skip-tags-wait-for-osds` or any name which suits test scenario.

**Comment by Ameenasuhani** [comment]
tier-2_upgrade_okr-bz-1876447-rpm.yaml can be changed  `**ceph-custom-name*`

**Review by Ameenasuhani — COMMENTED** [review]
LGTM

**Comment by rahullepakshi** [comment]
@sunilkumarn417 @psathyan @veera-raghava-reddy, please review

**Comment by rahullepakshi** [comment]
logs-

At 4.2, tier-2_upgrade_skip-tags-wait-for-osds.yaml - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1IMN4C/
FYI - We see a failure in upgrade since 4.2 is already GAed and no new builds available so far,  the comparison of versions failed the test case.

At 4.3, I see rhos-d authentication error to run the test. Once resolved, I will attach the logs

**Review by psathyan — DISMISSED** [review]

**Review by Ameenasuhani — DISMISSED** [review]

**Review by sunilkumarn417 — DISMISSED** [review]
LGTM. we could merge this PR after successful execution.

**Comment by psathyan** [comment]
@rahullepakshi please resolve the conflicts

**Comment by rahullepakshi** [comment]
Test run log at 4.3- http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-8L53MC/

**Review by psathyan — APPROVED** [review]

**Pull request merged**
